src/rotativeperformancelogger.cpp

```diff
@@ -0,0 +1,812 @@
+    std::mutex mLogMutex;
+    std::mutex mLogRotationMutex;
+    LogLinkedList mLogListFirst;
+    LogLinkedList* mLogListLast = &mLogListFirst;
+    bool mLogExit = false;
+    bool mFlushLog = false;
+    bool mCloseLog = false;
+    bool mForceRenew = false; //to force removal of all logs and create an empty new log
+    int mFlushOnLevel = MegaApi::LOG_LEVEL_WARNING;
+    std::chrono::seconds mLogFlushPeriod = std::chrono::seconds(10);
+    std::chrono::steady_clock::time_point mNextFlushTime = std::chrono::steady_clock::now() + mLogFlushPeriod;
+    MegaFileSystemAccess * mFsAccess;
+    ArchiveType mArchiveType = archiveTypeTimestamp;
+    long int archiveMaxFileAgeSeconds = 30 * 86400; // one month
+
+    friend RotativePerformanceLogger;
+
+public:
+    RotativePerformanceLoggerLoggingThread()
+    {
+        mFsAccess = new MegaFileSystemAccess();
+    }
+
+    ~RotativePerformanceLoggerLoggingThread()
+    {
+        delete mFsAccess;
+    }
+
+    void startLoggingThread(const LocalPath& logsPath, const LocalPath& fileName)
+    {
+        if (!mLogThread)
+        {
+            mLogThread.reset(new std::thread([this, logsPath, fileName]() {
+                logThreadFunction(logsPath, fileName);
+            }));
+        }
+    }
+
+    void log(int loglevel, const char *message, const char **directMessages = nullptr, size_t *directMessagesSizes = nullptr, int numberMessages = 0);
+
+    static void gzipCompressOnRotate(LocalPath localPath, LocalPath destinationLocalPath)
+    {
+        MegaFileSystemAccess * fsAccess = new MegaFileSystemAccess();
+
+        std::ifstream file(localPath.localpath.c_str(), std::ofstream::out);
+        if (!file.is_open())
+        {
+            std::cerr << "Unable to open log file for reading: " << localPath.toPath(*fsAccess) << std::endl;
+            delete fsAccess;
+            return;
+        }
+
+        auto gzdeleter = [](gzFile_s* f) { if (f) gzclose(f); };
+    #ifdef _WIN32
+        std::unique_ptr<gzFile_s, decltype(gzdeleter)> gzfile{ gzopen_w(destinationLocalPath.localpath.c_str(), "wb"), gzdeleter};
+    #else
+        std::unique_ptr<gzFile_s, decltype(gzdeleter)> gzfile{ gzopen(destinationLocalPath.localpath.c_str(), "wb"), gzdeleter };
+    #endif
+        if (!gzfile)
+        {
+            std::cerr << "Unable to open gzfile for writing: " << localPath.toPath(*fsAccess) << std::endl;
+            delete fsAccess;
+            return;
+        }
+
+        std::string line;
+        while (std::getline(file, line))
+        {
+            line.push_back('\n');
+            if (gzputs(gzfile.get(), line.c_str()) == -1)
+            {
+                std::cerr << "Unable to compress log file: " << localPath.toPath(*fsAccess) << std::endl;
+                return;
+            }
+        }
+
+        fsAccess->unlinklocal(localPath);
+        delete fsAccess;
+    }
+
+private:
+    LocalPath logArchiveNumbered_getFilename(LocalPath baseFileName, int logNumber)
```

Comment:
I believe all these functions could use `const LocalPath &` instead of `LocalPath`

Reply:
Yes. Actually the only non const usage was for dopen and dnext. I will use a copy of the variable for this couple of calls.